res/gamedata/scripts/ui_load_dialog.script

```diff
@@ -154,6 +154,10 @@ function file_data(fname)
 		local file		=	flist:GetAt(0)
 		local sg = CSavedGameWrapper(fname)
 
+		local isSaveValid = sg:valid_save()
+		self.load_btn:Enable(isSaveValid)
```

Comment:
I think we shouldn't allow this function to modify state of the UI. The function is designed for parsing save data, so by editing the dialog state we violate the single-responsibility principle. I suggest you to return from this function the state of parsing data (true for successful parsing, false otherwise) and process the result in load_dialog:OnListItemClicked.